typescript/packages/wallets/sui/src/SuiKeyPairWalletClient.ts

```diff
@@ -0,0 +1,58 @@
+import type { Signature } from "@goat-sdk/core";
+import { SuiClient, type SuiTransactionBlockResponse } from "@mysten/sui.js/client";
+import { type Ed25519Keypair } from "@mysten/sui.js/keypairs/ed25519";
+import { TransactionBlock } from "@mysten/sui.js/transactions";
+import { SuiWalletClient } from "./SuiWalletClient";
+import type { AwesomeChainResponse, SuiQuery, SuiTransaction, Transaction } from "./types";
+
+export type SuiKeyPairWalletClientParams = {
+    client: SuiClient;
+    keypair: Ed25519Keypair;
+};
+
+export class SuiKeyPairWalletClient extends SuiWalletClient {
+    private keypair: Ed25519Keypair;
+
+    constructor(params: SuiKeyPairWalletClientParams) {
+        super({ client: params.client });
+        this.keypair = params.keypair;
+    }
+
+    async sendTransaction(transaction: SuiTransaction): Promise<Transaction> {
+        const txb = new TransactionBlock();
+        // Create a transfer transaction
+        txb.transferObjects([txb.gas], txb.pure(transaction.to));
```

Comment:
This sendTransaction function doesn't allow you to interact with arbitrary contracts and parameters. You need to make it generic.

One approach would be to pass in the Transaction object directly as parameter inside the SuiTransaction.

`transaction.transaction`